drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,313 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+		}
+
+		//Get the node where the vm is scheduled after the migration
+		nodeNameAfterMigration, err := GetNodeOfVM(vms[0])
+		if err != nil {
+			return "", false, err
+		}
+		if nodeName == nodeNameAfterMigration {
+			return "", true, fmt.Errorf("VM pod live migrated [%s] in namespace [%s] but is still on the same node [%s]", testPod.Name, testPod.Namespace, nodeName)
+		}
+		log.InfoD("VM pod live migrated to node: [%s] in namespace [%s] is in running state", nodeNameAfterMigration, testPod.Namespace)
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func getVMPod(virtualMachineCtx *scheduler.Context, vol *volume.Volume) (*corev1.Pod, error) {
+	pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pods for volume %s of context %s: %w", vol.ID, virtualMachineCtx.App.Key, err)
+	}
+
+	var found corev1.Pod
+	for _, pod := range pods {
+		if pod.Labels["kubevirt.io"] == "virt-launcher" && pod.Status.Phase == corev1.PodRunning {
+			if found.Name != "" {
+				// there should be only one VM pod in the running state (otherwise live migration is in progress)
+				return nil, fmt.Errorf("more than 1 KubeVirt pods (%s, %s) are in running state for volume %s",
+					found.Name, pod.Name, vol.ID)
+			}
+			found = pod
+		}
+	}
+	if found.Name == "" {
+		return nil, fmt.Errorf("failed to find a running pod for volume %s", vol.ID)
+	}
+	return &found, nil
+}
+
+func IsVMBindMounted(virtualMachineCtx *scheduler.Context, driver scheduler.Driver, volumeDriver volume.Driver, initialCheck bool) (bool, error) {
+	vols, err := driver.GetVolumes(virtualMachineCtx)
+	if err != nil {
+		return false, err
+	}
+	if len(vols) == 0 {
+		return false, fmt.Errorf("no volumes found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+
+	// Get the node where pod is scheduled
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{virtualMachineCtx})
+	if err != nil {
+		return false, err
+	}
+	if len(vms) != 1 {
+		return false, fmt.Errorf("expected 1 VM for VM [%s] in namespace [%s] but got [%d]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace, len(vms))
+	}
+	vm := vms[0]
+	vmNodeName, err := GetNodeOfVM(vm)
+	if err != nil {
+		return false, err
+	}
+	log.Infof("VM [%s] is deployed on node [%s]", virtualMachineCtx.App.Key, vmNodeName)
+
+	// Keep a track of replicaset and consider this as the source of truth
+	volInspect, err := volumeDriver.InspectVolume(vols[0].ID)
+	if err != nil {
+		return false, fmt.Errorf("failed to inspect volume [%s]: %w", vols[0].ID, err)
+	}
+	globalReplicSet := volInspect.ReplicaSets
+	log.InfoD("Length of replicaset: %d", len(globalReplicSet))
+
+	// Check if the volume is mounted on the node
+	for _, vol := range vols {
+		// Check if the volume is mounted on the node
```

Comment:
for readability, consider moving the code inside the for loop in a separate function that checks only one volume

Reply:
Moved the logic inside the for loop to another function named IsVolumeBindMounted